pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -326,11 +327,44 @@ func (c *Cluster) postReconcileUpdateOSDProperties(desiredOSDs map[int]*OSDInfo)
 	}
 	logger.Debugf("post processing osd properties with %d actual osds from ceph osd df and %d existing osds found during reconcile", len(osdUsage.OSDNodes), len(desiredOSDs))
 	for _, actualOSD := range osdUsage.OSDNodes {
-		if desiredOSD, ok := desiredOSDs[actualOSD.ID]; ok {
-			if err := c.updateDeviceClassIfChanged(actualOSD.ID, desiredOSD.DeviceClass, actualOSD.DeviceClass); err != nil {
-				// Log the error and allow other updates to continue
-				logger.Error(err)
-			}
+		if err := c.resizeOsdCrushWeight(actualOSD); err != nil {
+			// Log the error and allow other updates to continue
+			logger.Error(fmt.Errorf("failed to resize osd crush weight on cluster in namespace %s %+v", c.clusterInfo.Namespace, err))
```

Comment:
@travisn in case of reweight not happen, do we document for manual operator re-start do to get the osd updated correctly?

Reply:
```suggestion
			logger.Errorf("failed to resize osd crush weight on cluster in namespace %s. %+v", c.clusterInfo.Namespace, err)
```

Reply:
> @travisn in case of reweight not happen, do we document for manual operator re-start do to get the osd updated correctly?

The impact of the reweight failing is minimal. If it succeeds on the next reconcile (even a few days later), it will likely be ok. However, if we find a scenario that has a common reason for failure, we may want to document it.